recordset/recordset.js

```diff
@@ -284,142 +278,81 @@ angular.module('recordset', ['ermrestjs', 'chaise.navbar', 'chaise.utils', 'chai
+    }
 
-            // Find shortest Key, used for paging and linking
-            var keys = table.keys.all().sort( function(a, b) {
-                return a.colset.length() - b.colset.length();
-            });
-            recordsetModel.keycols = keys[0].colset.columns;
-
-            // sorting
-            var sort = [];
-
-            // user selected column as the priority in sort
-            // followed by all the key columns
-            if (context.sort !== null) {
-                if (context.sort.endsWith("::desc::")) {
-                    recordsetModel.sortby = decodeURIComponent(
-                        context.sort.match(/(.*)::desc::/)[1]
-                    );
-                    recordsetModel.sortOrder = 'desc';
-                } else {
-                    recordsetModel.sortby = decodeURIComponent(context.sort);
-                    recordsetModel.sortOrder = 'asc';
-                }
-
-                // this will cause program to throw exception is sort column is not valid
-                table.columns.get(recordsetModel.sortby);
-
-                sort.push({"column": recordsetModel.sortby, "order": recordsetModel.sortOrder});
-            }
-
-            for (i = 0; i < recordsetModel.keycols.length; i++) { // all the key columns
-                var col = recordsetModel.keycols[i].name;
-                if (col !== recordsetModel.sortby) {
-                    sort.push({"column": col, "order": "asc"});
-                }
-            }
-
-            // first get row count
-            table.entity.count(recordsetModel.filter).then(function (count) {
-                recordsetModel.count = count;
-
-                // get rowset from table
-                table.entity.get(recordsetModel.filter, pageInfo.pageLimit, null, sort).then(function (rowset) {
-                    console.log(rowset);
-                    recordsetModel.rowset = rowset;
-
-                    pageInfo.loading = false;
-                    pageInfo.recordStart = 1;
-                    pageInfo.recordEnd = pageInfo.recordStart + rowset.length() - 1;
-                    pageInfo.previousButtonDisabled = true;
-                    pageInfo.nextButtonDisabled = recordsetModel.count <= pageInfo.recordEnd;
-
-                }, function (error) {
-                    console.log(error);
-                    pageInfo.loading = false;
-                    pageInfo.previousButtonDisabled = true;
-                    pageInfo.nextButtonDisabled = true;
-
-                    if (error instanceof ERMrest.UnauthorizedError) {
-                        // session has expired, login
-                        Session.login($window.location.href);
-                    }
-                });
-            }, function (error) {
-                pageInfo.loading = false;
-                pageInfo.previousButtonDisabled = true;
-                pageInfo.nextButtonDisabled = true;
+    var ermrestUri = UriUtils.chaiseURItoErmrestURI($window.location);
```

Comment:
In my branch, `chaise-to-ermrest-uri`, I made changes to this function so that it throws errors when catalog or table/schema are not specified. You probably want to wrap this in a `try/catch`.